internal/ingress/controller/kong.go

```diff
@@ -114,6 +119,91 @@ func (n *NGINXController) OnUpdate(ingressCfg *ingress.Configuration) error {
 	return nil
 }
 
+func (n *NGINXController) syncGlobalPlugins() error {
+	glog.Infof("syncing global plugins")
+
+	targetGlobalPlugins, err := n.store.ListGlobalKongPlugins()
+	if err != nil {
+		return err
+	}
+	targetPluginMap := make(map[string]*pluginv1.KongPlugin)
+	var duplicates []string // keep track of duplicate
+
+	for i := 0; i < len(targetGlobalPlugins); i++ {
+		name := targetGlobalPlugins[i].Name
+		// empty name skip it
+		if name == "" {
+			continue
+		}
+		if _, ok := targetPluginMap[name]; ok {
+			glog.Error("Multiple KongPlugin definitions found with 'global' annotation for :", name,
+				", the plugin will not be applied")
+			duplicates = append(duplicates, name)
+			continue
+		}
+		targetPluginMap[name] = targetGlobalPlugins[i]
+	}
+
+	// remove duplicates
+	for _, plugin := range duplicates {
+		delete(targetPluginMap, plugin)
+	}
+
+	client := n.cfg.Kong.Client
+	plugins, err := client.Plugins().List(nil)
+	if err != nil {
+		return err
+	}
+
+	// plugins in Kong
+	currentGlobalPlugins := make(map[string]kongadminv1.Plugin)
+	for _, plugin := range plugins.Items {
+		if plugin.Route == "" && plugin.Service == "" && plugin.Consumer == "" {
```

Comment:
Does it make sense to have a flag indicating it's global plugin in the `plugin` structure instead of depending on the assertion that "it's not applied to any entities so it's global"?